### PR TITLE
add redpar preparation for Joerg new processing routine

### PR DESCRIPTION
**Pull request by Kvieta1990**
- To implement the preparation of `.redpar` file to work with the new sum scan script by Joerg.

- The preparation of `.redpar` file happens at the stage of creating sum scan output file.

- All information needed by `.redpar` file is already there in the GUI and we just need grab them and fill into the `.redpar` file.

- A separate `.redpar` file is needed for each checked row in the preparation table under `Post Processing` tab.

- File name for each `.redpar` should be set to the `Name` property in the preparation table under `Post Processing` tab.

### CODE REVIEW

File: addie/processing/idl/run_sum_scans.py
Comment (marshallmcdonnell) [inline addie/processing/idl/run_sum_scans.py:164]: So curious, why not add the name, formula, mass density, radius, packing fraction, and geometry as keys in the `_runs` dictionary, instead of returning now a list?

For the dictionary, the key lets you know some sort of metadata about the value in the dictionary.
For a list, the person has to know the index, which really contains no metadata for that entry.

File: addie/processing/idl/run_sum_scans.py
Comment (marshallmcdonnell) [inline addie/processing/idl/run_sum_scans.py:40]: See comment below about the change from a dict to a list return type. May not need all these?

**Comment by Kvieta1990** [comment]
> I don't know what the redpar file is (maybe provide an example?) but seems okay to me.
> Yet, does there need to be a change to the sum scans script we use in ADDIE?

The `redpar` file is just a input file needed by Joerg's new routine of doing the sum scan and I don't think there is more than the preparation of this `redpar` file that needs to be further implemented. Here you can find an example,

`/SNS/NOM/IPTS-25335/shared/autoNOM_SiO2_in_shifter_test/diamond.redpar`

File: addie/processing/idl/run_sum_scans.py
Comment (Kvieta1990) [reply addie/processing/idl/run_sum_scans.py]: new codes pushed to address this issue

File: addie/processing/idl/run_sum_scans.py
Comment (Kvieta1990) [reply addie/processing/idl/run_sum_scans.py]: see comment above. Try to change the data structure here to dictionary.

File: addie/processing/idl/run_sum_scans.py
Comment (marshallmcdonnell) [inline addie/processing/idl/run_sum_scans.py:108]: Two things here I am just noticing now:
 1) The file should be opened using a `with` statement, current implementation is actually an anti-pattern (https://docs.quantifiedcode.com/python-anti-patterns/maintainability/not_using_with_to_open_files.html)

 2) I was hoping we could get the example redpar file here so maybe we could write an actual test to make sure this works since right now, I have no idea if it works or not. It probably would help to break this up into two different parts: one that creates the output but doesn't write it to file and then another part that opens the file and writes the content. This would help since the first part could be a method / function that you write a normal test for where you check the contents of the output data structure for the redpar file and the second could be where you check the written file against a redpar target test file we keep in the repository. The current way it is written, everything is embedded into a larger function that we cannot test to make sure it works (or that we don't break it in the future when we move the table around)

Comment (Kvieta1990) [reply addie/processing/idl/run_sum_scans.py]: > Two things here I am just noticing now:
> 
> 1. The file should be opened using a `with` statement, current implementation is actually an anti-pattern (https://docs.quantifiedcode.com/python-anti-patterns/maintainability/not_using_with_to_open_files.html)
> 2. I was hoping we could get the example redpar file here so maybe we could write an actual test to make sure this works since right now, I have no idea if it works or not. It probably would help to break this up into two different parts: one that creates the output but doesn't write it to file and then another part that opens the file and writes the content. This would help since the first part could be a method / function that you write a normal test for where you check the contents of the output data structure for the redpar file and the second could be where you check the written file against a redpar target test file we keep in the repository. The current way it is written, everything is embedded into a larger function that we cannot test to make sure it works (or that we don't break it in the future when we move the table around)

- Codes cleaned up to cope with the anti-pattern issue.

- Routine separated into two parts and implemented as functions for each part. I am not sure how to write a test for this. Could you please help with this? Thank you @marshallmcdonnell!

Comment (marshallmcdonnell) [reply addie/processing/idl/run_sum_scans.py]: I think you have already done plenty to make it testable, thanks!
We can just push it to another PR but the structure is now much better to write a test for it.

Comment (Kvieta1990) [reply addie/processing/idl/run_sum_scans.py]: No problem at all! So this PR should be ready to be merged, right? I have done the test locally on analysis and everything is working perfectly.

**Comment by marshallmcdonnell** [comment]
> No problem at all! So this PR should be ready to be merged, right? I have done the test locally on analysis and everything is working perfectly.

Yep, I'll approve after I see the green from CI :+1:

**Review by marshallmcdonnell — APPROVED** [review]

**Comment by marshallmcdonnell** [comment]
@peterfpeterson do you still want to review? If not, I am fine to merge this.

**Comment by Kvieta1990** [comment]
> @peterfpeterson do you still want to review? If not, I am fine to merge this.

@peterfpeterson sorry, pete I did not wait until you approve the PR, but I think everything is Ok with this PR considering @marshallmcdonnell's review and my own testing.

**Pull request merged**
